pkg/config/expand.go

```diff
@@ -166,6 +169,45 @@ func (bp Blueprint) expandBackend(grp *Group) {
 	}
 }
 
+func getDefaultGoogleProviders(bp Blueprint) map[string]TerraformProvider {
+	gglConf := Dict{}
+	for s, v := range map[string]string{
+		"project": "project_id",
+		"region":  "region",
+		"zone":    "zone"} {
+		if bp.Vars.Has(v) {
+			gglConf = gglConf.With(s, GlobalRef(v).AsValue())
+		}
+	}
+	return map[string]TerraformProvider{
+		"google": TerraformProvider{
+			Source:        "hashicorp/google",
+			Version:       ">= 4.84.0, < 5.32.0",
+			Configuration: gglConf},
+		"google-beta": TerraformProvider{
+			Source:        "hashicorp/google-beta",
+			Version:       ">= 4.84.0, < 5.32.0",
+			Configuration: gglConf}}
+}
+
+func (bp Blueprint) expandProviders(grp *Group) {
+	// 1. DEFAULT: use TerraformProviders provider dictionary (if supplied)
+	// 2. If top-level TerraformProviders is defined, insert that
+	//    provider dictionary into resource groups which have no explicit
+	//    TerraformProviders
+	defaults := bp.TerraformProviders
+	pv := &grp.TerraformProviders
+	if defaults == nil {
+		defaults = getDefaultGoogleProviders(bp)
+	}
+	if (*pv) == nil {
+		(*pv) = map[string]TerraformProvider{}
```

Comment:
this is essentially a copy, is there a reason to write it as 4 lines instead of 1?

Reply:
Fixed